### DISCUSSION TITLE
Email sync feature for Online Bank Statement Sync

### DISCUSSION THREAD

**Discussion by mayankb2103** [moneymanagerex/moneymanagerex#4821] [Ideas]
Hi, 
I was looking for a Money Manager where it does automatically sync my Online bank account transactions. When I kinda stumbled on this app. I really liked the idea of the app and the fact that it is open source. I saw some discussion going on about [Online bank sync](https://github.com/moneymanagerex/moneymanagerex/issues/2119) here where someone suggested using Bank API Integration. But in my opinion, that is not scalable for syncing bank statements across banks in different countries. Where somewhere in a country like India rules are stringent and highly likely that such API might not exist for many banks. 

I have an idea, what if with the user's consent we support syncing bank statements from Monthly Email statements. We can just ask for read permission from the user (with the email id and specific date of bank statement) and will sync the statement and import transactions. The transaction format file will be mostly PDF, which we can reconcile to support online Sync?
Any other suggestions and enhancements are welcome towards this

**Comment by tactilis** [comment]
@mayankb2103
On the face of it, your idea may seem like a good solution, but in fact is full of difficulties.

Some thoughts on the requirements:

> We can just ask for read permission from the user (with the email id and specific date of bank statement)

This implies that MMEX will need to interface with the user's email account. That's not a simple task; different accounts have different authentication requirements and different protocols (e.g. IMAP, POP3, etc) for accessing the emails. Your simple statement _"We can just ask for read permission"_ is hiding a great deal of complexity and has trust and security implications.

Then there is the question of how MMEX would locate the email. You suggest by date but there may be many emails on that date, so now it's necessary to work out which is the one from the bank and contains a statement.

> The transaction format file will be mostly PDF

_"Mostly PDF?"_  So not always?  There's lots of hidden complexity in those two words.

And you are asking for MMEX to be able to parse the contents of the PDF document and work out which fields are which in the statement. Parsing PDFs can be done but is not easy; the format was never designed for parsing; it is a reading and print format.

Every bank will have a different statement layout, and the layout will be subject to change at any time if the bank redesigns its statements.
 
 
You idea is a non-starter I'm afraid. It's a huge amount of work to implement, and then maintain, for very little benefit.  

> Any other suggestions and enhancements are welcome towards this

Yes. Please familiarise yourself with the existing 'Import CSV' and 'Import QIF' functionality.

It's easy to download a CSV or QIF file from banks and import the transactions into MMEX. There is already a flexible user interface for handling the variations in the file format produced by different banks.
 
 
I've written at length above, not to shoot you down, but to try to show that an apparently 'simple' feature request can be hugely complex and have implications far beyond what you considered.

If you need any help with Import CSV/Import QIF, then please do ask.
